lib/LANraragi/Plugin/Metadata/Eze.pm

```diff
@@ -42,86 +42,60 @@ sub get_tags {
     my ($save_title) = @_;    # Plugin parameters
 
     my $logger = get_plugin_logger();
+
     my $path_in_archive = is_file_in_archive( $lrr_info->{file_path}, "info.json" );
 
-    my @suffixlist = qw(.zip .cbz .rar .cbr .tar.gz .cbt .lzma .7z .cb7 .xz .pdf .epub);
-    my ($name, $path, $suffix) = fileparse($lrr_info->{file_path}, @suffixlist);
+    my ($name, $path, $suffix) = fileparse($lrr_info->{file_path}, qr/\.[^.]*/);
     my $path_nearby_json = $path . $name . '.json';
 
-    if ($path_in_archive) {
-
-        #Extract info.json
-        my $filepath = extract_file_from_archive( $lrr_info->{file_path}, $path_in_archive );
+    my $filepath = undef;
```

Comment:
The critic CI shouldve yelled about this too (looks like its checks API integration is broken oops), but there's no need to initialize variables to `undef`:  

```
my $filepath;
my $delete_after_parse;
```